docs/contentScript.js

```diff
@@ -88,35 +91,77 @@
 
 })();
 
-const GRADIENT_LENGTH_PERCENTAGE = 0.5;
-
-var ps = document.getElementsByTagName('p');
-for(let i=0; i<ps.length; i++){
-	var p = ps[i];
-	var lines = lineWrapDetector.getLines(p);
-	for(let j=0; j<lines.length; j++){
-		let line = lines[j];
-		let red = 0, blue = 0, green = 0;
-		for(let k=0; k<line.length; k++){
-			switch(j%4){
-				case 0: //Blue ending
-					blue = k>(line.length-1)*GRADIENT_LENGTH_PERCENTAGE ? getGradient(k, line.length) : 0;
-					break;
-				case 1: //Blue beginning
-					blue = k<(line.length-1)*GRADIENT_LENGTH_PERCENTAGE ? getGradient(k, line.length) : 0;
-					break;
-				case 2: //Red ending
-					red = k>(line.length-1)*GRADIENT_LENGTH_PERCENTAGE ? getGradient(k, line.length) : 0;
-					break;
-				case 3: //Red beginning
-					red = k<(line.length-1)*GRADIENT_LENGTH_PERCENTAGE ? getGradient(k, line.length) : 0;
-					break;
+(function() {
+'use strict';
+
+// Linear interpolate between v0 and v1 at percent t
+function lerp(v0, v1, t)
+{
+	return v0 * (1 - t) + v1 * t
+}
+
+// Convert a hex triplet (#XXXXXX) to an array containing red, green, and blue
+function hex_to_rgb(hex)
+{
+	return hex.replace('#', '').match(/.{1,2}/g).map(
+		x => parseInt(x, 16)
+	);
+}
+
+// Color all lines in the page
+function applyGradient(colors, color_text, gradient_size)
+{
+	const paragraphs = document.getElementsByTagName('p');
+	const base_color = hex_to_rgb(color_text);
+	let coloridx = 0;
+	let lineno = 0;
+
+	for (let paragraph of paragraphs) {
+		const lines = lineWrapDetector.getLines(paragraph);
+
+		for (let line of lines) {
+			// Alternate between left and right for every color
+			const active_color = hex_to_rgb(colors[coloridx]);
+
+			// Flip array around if on left to color correctly
+			const is_left = (lineno % 2 === 0);
+			if(is_left) {
+				line = Array.from(line).reverse();
 			}
-			line[k].style.color="rgb("+red+","+green+","+blue+")";
+
+			// Color lines using lerp of RGB values
+			for (let loc in line) {
+				const t = 1 - (loc / (line.length * gradient_size / 50));
+				const red = lerp(base_color[0], active_color[0], t);
+				const green = lerp(base_color[1], active_color[1], t);
+				const blue = lerp(base_color[2], active_color[2], t);
+
+				line[loc].style.color = "rgb(" + (red|0) + "," + (green|0) + "," + (blue|0) + ")";
+			}
+
+			// Increment color index after every left/right pair, and lineno
+			// after every line
+			if (!is_left) {
+				coloridx = (coloridx + 1) % colors.length;
+			}
+			lineno += 1;
 		}
 	}
 }
 
-function getGradient(k, len){
-	return Math.round(Math.abs(255*(((k/(len-1))-GRADIENT_LENGTH_PERCENTAGE)/GRADIENT_LENGTH_PERCENTAGE)));
-}
+// Listen for messages in background script
+chrome.runtime.onMessage.addListener((message) => {
```

Comment:
This file is used in wasplinereader.com to display the gradients, given that a background message won't ever reach it (it's a standalone webpage), the gradients will never be applied on this page.
With the introduction of your changes, maybe we should revert it back to its previous state and remove the lines that modify it from build.sh

Reply:
Actually that would be a better solution than modifying the .gitignore. Let's do that then.